src/platforms/ctxlink/platform.h

```diff
@@ -172,15 +178,48 @@
 #define IRQ_PRI_USBUSART     (2U << 4U)
 #define IRQ_PRI_USBUSART_DMA (2U << 4U)
 #define IRQ_PRI_TRACE        (0U << 4U)
+#define IRQ_PRI_SWO_DMA      (0U << 4U)
 
-/* Use TIM3 Input 2 (from PC7/TDO) AF2, trigger on Rising Edge */
-#define TRACE_TIM          TIM3
-#define TRACE_TIM_CLK_EN() rcc_periph_clock_enable(RCC_TIM3)
-#define TRACE_IRQ          NVIC_TIM3_IRQ
-#define TRACE_ISR(x)       tim3_isr(x)
-#define TRACE_IC_IN        TIM_IC_IN_TI2
-#define TRACE_TRIG_IN      TIM_SMCR_TS_TI1FP1
-#define TRACE_TIM_PIN_AF   GPIO_AF2
+#define PLATFORM_HAS_TRACESWO
+#define NUM_TRACE_PACKETS 256U /* 16K buffer */
+
+#if TRACESWO_PROTOCOL == 1
+
+/* Use TIM3 Input 2 from PC7/TDO, AF2, trigger on Rising Edge */
+#define TRACE_TIM TIM3
+#define TRACE_TIM_CLK_EN()
+#define TRACE_IRQ             NVIC_TIM3_IRQ
+#define TRACE_ISR(x)          tim3_isr(x)
+#define TRACE_IC_IN           TIM_IC_IN_TI1
```

Comment:
Isn't this Timer Input 1 (PC6)? I guess it should be TI2 (PC7) but I haven't tested this mode.

Reply:
It is, however he had a conversation about us and agreed to leave this as is and we'll fix it as part of the Manchester mode SWO branch as due to needing CH2, it needs some other things fixed in the SWO implementation for it to actually work.